### DISCUSSION TITLE
Why does ResourceStorageService.AddAsync(...) add then get the same resource immediately after?

### DISCUSSION THREAD

**Discussion by afij** [FirelyTeam/spark#745] [Q&A] [answered]
I have been reviewing the code and noticed that when the following code is ran, you insert the entry into the database and then immediately search for it again?

https://github.com/FirelyTeam/spark/blob/9434518bb9f906b76e51951e8d966afcf508e618/src/Spark.Engine/Service/FhirServiceExtensions/ResourceStorageService.cs#L41C1-L60C10

Comparing the Entry being inserted and then the Result retrieved from the database, they don't look that different but I might just be missing something important. Just curious as to what the purpose is. Hope that a "discussion" is the right place to ask a question like this.

**Comment by kennethmyhra** [comment, marked as answer]
See #1138
